### PR TITLE
Add Markdown fields to app forms

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add Markdown fields to app forms. Also piggybacks a fix to remove readonly fields from commands.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-31052
https://mattermost.atlassian.net/browse/MM-36043

#### Related Pull Requests
Mobile: https://github.com/mattermost/mattermost-mobile/pull/5407
Apps Framework: 

#### Screenshots
Screenshot from 2021-05-27 11-42-26
Screenshot from 2021-05-27 11-42-38
Screenshot from 2021-05-27 11-56-39

#### Release Note
```release-note
Apps can now add arbitrary markdown in between fields on forms.
```

### CODE REVIEW

File: components/suggestion/command_provider/app_command_parser/app_command_parser.ts [hunk 1]
```diff
@@ -839,7 +900,7 @@ export class AppCommandParser {
     // getChannel gets the channel in which the user is typing the command
     private getChannel = (): Channel | null => {
         const state = this.store.getState();
-        return getChannel(state, this.channelID);
+        return selectChannel(state, this.channelID);
```
Comment (hmhealey) [inline components/suggestion/command_provider/app_command_parser/app_command_parser.ts:903]: Definitely 0/5 on this, but I've been thinking of changing the actions to load objects from the server to be `fetchX` and leaving the selectors as `getX`. No actual plans to do that though

**Review by hmhealey — APPROVED** [review]

File: components/suggestion/command_provider/app_command_parser/app_command_parser.ts (hunk 1 above)
Comment (mickmister) [reply components/suggestion/command_provider/app_command_parser/app_command_parser.ts:903]: I agree, `fetch` and `get` make the most sense to me

**Review by mickmister — APPROVED** [review]
LGTM, nice work!

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- All Markdown is displaying correctly
- Tested on Browser and desktop
- Tested on mobile web view
- No JS error in console when modal opens or on submit

LGTM!
Added a Zephyr test to cover this _MM-T4106_

**Comment by DHaussermann** [comment]
@larkox I'm seeing an issue that was maybe introduced by this PR. I see the following error when trying to open the markdown modal `Cannot read property 'map' of undefined` I'll follow-up to see what our best option is here.

**Pull request merged**